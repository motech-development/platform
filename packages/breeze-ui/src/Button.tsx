import React, { FC } from 'react';
import styled from 'styled-components';

const BaseButton = styled.button`
  appearance: none;
  vertical-align: middle;
  user-select: none;
  white-space: nowrap;
`;

export interface IButtonProps {
  type?: 'submit' | 'reset' | 'button';
}

const Button: FC<IButtonProps> = ({ children, type = 'button' }) => {
  return <BaseButton type={type}>{children}</BaseButton>;
};

export default Button;
