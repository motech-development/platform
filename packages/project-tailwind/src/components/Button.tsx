import { ElementType } from 'react';
import { Box, PolymorphicComponentProps } from 'react-polymorphic-box';
import {
  Sizing,
  Themes,
  TSizing,
  TTheme,
  useTailwind,
} from '../utilities/tailwind';

/** Default component element */
const DEFAULT_ELEMENT = 'button';

/** Button component props */
export type TButtonProps<E extends ElementType> = PolymorphicComponentProps<
  E,
  {
    /** Renders component as a block element */
    block?: boolean;

    /** Component theme */
    theme?: TTheme;

    /** Component size */
    size?: TSizing;
  }
>;

/**
 * Flexible call to action button
 *
 * @param props - Button component props
 *
 * @returns Button component
 */
export function Button<E extends ElementType = typeof DEFAULT_ELEMENT>({
  block = false,
  className,
  disabled,
  theme = Themes.PRIMARY,
  size = Sizing.MD,
  ...rest
}: TButtonProps<E>) {
  const { createStyles } = useTailwind(theme, size);

  const buttonStyles = createStyles({
    classNames: [
      {
        'cursor-not-allowed bg-opacity-75 hover:bg-opacity-75': disabled,
        'w-full': block,
      },
      'group items-center relative shadow-2xl inline-flex font-display justify-center font-medium transition-colors select-none focus:outline-none focus:ring-2 focus:ring-offset-2',
      className,
    ],
    sizing: {
      lg: ['py-4 px-6 text-lg'],
      md: ['py-2 px-4 text-base'],
      none: ['p-0 text-sm'],
      sm: ['py-0 px-2 text-sm'],
    },
    theme: {
      danger: [
        'bg-red-600 text-white',
        {
          'hover:bg-red-700 focus:ring-red-500': !disabled,
        },
      ],
      primary: [
        'bg-blue-600 text-white',
        {
          'hover:bg-blue-700 focus:ring-blue-500': !disabled,
        },
      ],
      secondary: [
        'bg-gray-100 text-gray-600',
        {
          'hover:bg-gray-300 focus:ring-gray-500': !disabled,
        },
      ],
      success: [
        'bg-green-600 text-white',
        {
          'hover:bg-green-700 focus:ring-green-500': !disabled,
        },
      ],
      warning: [
        'bg-yellow-600 text-white',
        {
          'hover:bg-yellow-700 focus:ring-yellow-500': !disabled,
        },
      ],
    },
  });

  return (
    <Box
      className={buttonStyles}
      as={DEFAULT_ELEMENT}
      {...{ disabled }}
      // eslint-disable-next-line react/jsx-props-no-spreading
      {...rest}
    >
      Test
    </Box>
  );
}
