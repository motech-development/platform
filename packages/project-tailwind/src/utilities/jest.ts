/* eslint-disable import/no-extraneous-dependencies */
import { render, RenderResult } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { UserEvent } from '@testing-library/user-event/dist/types/setup/setup';
import { ReactElement } from 'react';
import { Sizing, Themes } from './tailwind';

export const themes = [
  {
    theme: Themes.DANGER,
  },
  {
    theme: Themes.PRIMARY,
  },
  {
    theme: Themes.SECONDARY,
  },
  {
    theme: Themes.SUCCESS,
  },
  {
    theme: Themes.WARNING,
  },
];

export const sizing = [
  {
    size: Sizing.LG,
  },
  {
    size: Sizing.MD,
  },
  {
    size: Sizing.NONE,
  },
  {
    size: Sizing.SM,
  },
];

type TSetup = RenderResult & {
  user: UserEvent;
};

/**
 * Combines Testing Library render function with user events setup
 *
 * @param ui - React element to test
 *
 * @returns Rendered component and test user
 */
export function setup(ui: ReactElement): TSetup {
  return {
    user: userEvent.setup({
      delay: null,
    }),
    ...render(ui),
  };
}
