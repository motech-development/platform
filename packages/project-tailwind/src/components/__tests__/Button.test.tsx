import { render } from '@testing-library/react';
import { ElementType } from 'react';
import { sizing, themes } from '../../utilities/jest';
import { Button } from '../Button';

const elements = [
  {
    attrs: {
      href: 'https://motech-development.github.io/platform/',
    },
    element: 'a' as ElementType,
  },
  {
    attrs: {
      type: 'button',
    },
    element: 'button' as ElementType,
  },
];

describe('Button', () => {
  describe.each(elements)('as element "$element"', ({ attrs, element }) => {
    describe.each(themes)('$theme', ({ theme }) => {
      it('should render the correct output when disabled', () => {
        const { asFragment } = render(
          // eslint-disable-next-line react/jsx-props-no-spreading
          <Button disabled as={element} theme={theme} {...attrs}>
            Hello, world
          </Button>,
        );

        expect(asFragment()).toMatchSnapshot();
      });

      it('should render the correct output when block element', () => {
        const { asFragment } = render(
          // eslint-disable-next-line react/jsx-props-no-spreading
          <Button block as={element} theme={theme} {...attrs}>
            Hello, world
          </Button>,
        );

        expect(asFragment()).toMatchSnapshot();
      });

      it.each(sizing)(
        'should render the correct output when size is $size',
        ({ size }) => {
          const { asFragment } = render(
            // eslint-disable-next-line react/jsx-props-no-spreading
            <Button as={element} size={size} theme={theme} {...attrs}>
              Hello, world
            </Button>,
          );

          expect(asFragment()).toMatchSnapshot();
        },
      );
    });
  });
});
