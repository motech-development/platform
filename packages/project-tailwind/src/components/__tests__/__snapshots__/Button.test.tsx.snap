// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button as element "a" danger should render the correct output when block element 1`] = `
<DocumentFragment>
  <a
    class="w-full group items-center relative shadow-2xl inline-flex font-display justify-center font-medium transition-colors select-none focus:outline-none focus:ring-2 focus:ring-offset-2 py-2 px-4 text-base bg-red-600 text-white hover:bg-red-700 focus:ring-red-500"
    href="https://motech-development.github.io/platform/"
  >
    Test
  </a>
</DocumentFragment>
`;

exports[`Button as element "a" danger should render the correct output when disabled 1`] = `
<DocumentFragment>
  <a
    class="cursor-not-allowed bg-opacity-75 hover:bg-opacity-75 group items-center relative shadow-2xl inline-flex font-display justify-center font-medium transition-colors select-none focus:outline-none focus:ring-2 focus:ring-offset-2 py-2 px-4 text-base bg-red-600 text-white"
    disabled=""
    href="https://motech-development.github.io/platform/"
  >
    Test
  </a>
</DocumentFragment>
`;

exports[`Button as element "a" danger should render the correct output when size is lg 1`] = `
<DocumentFragment>
  <a
    class="group items-center relative shadow-2xl inline-flex font-display justify-center font-medium transition-colors select-none focus:outline-none focus:ring-2 focus:ring-offset-2 py-4 px-6 text-lg bg-red-600 text-white hover:bg-red-700 focus:ring-red-500"
    href="https://motech-development.github.io/platform/"
  >
    Test
  </a>
</DocumentFragment>
`;

exports[`Button as element "a" danger should render the correct output when size is md 1`] = `
<DocumentFragment>
  <a
    class="group items-center relative shadow-2xl inline-flex font-display justify-center font-medium transition-colors select-none focus:outline-none focus:ring-2 focus:ring-offset-2 py-2 px-4 text-base bg-red-600 text-white hover:bg-red-700 focus:ring-red-500"
    href="https://motech-development.github.io/platform/"
  >
    Test
  </a>
</DocumentFragment>
`;

exports[`Button as element "a" danger should render the correct output when size is none 1`] = `
<DocumentFragment>
  <a
    class="group items-center relative shadow-2xl inline-flex font-display justify-center font-medium transition-colors select-none focus:outline-none focus:ring-2 focus:ring-offset-2 p-0 text-sm bg-red-600 text-white hover:bg-red-700 focus:ring-red-500"
    href="https://motech-development.github.io/platform/"
  >
    Test
  </a>
</DocumentFragment>
`;

exports[`Button as element "a" danger should render the correct output when size is sm 1`] = `
<DocumentFragment>
  <a
    class="group items-center relative shadow-2xl inline-flex font-display justify-center font-medium transition-colors select-none focus:outline-none focus:ring-2 focus:ring-offset-2 py-0 px-2 text-sm bg-red-600 text-white hover:bg-red-700 focus:ring-red-500"
    href="https://motech-development.github.io/platform/"
  >
    Test
  </a>
</DocumentFragment>
`;

exports[`Button as element "a" primary should render the correct output when block element 1`] = `
<DocumentFragment>
  <a
    class="w-full group items-center relative shadow-2xl inline-flex font-display justify-center font-medium transition-colors select-none focus:outline-none focus:ring-2 focus:ring-offset-2 py-2 px-4 text-base bg-blue-600 text-white hover:bg-blue-700 focus:ring-blue-500"
    href="https://motech-development.github.io/platform/"
  >
    Test
  </a>
</DocumentFragment>
`;

exports[`Button as element "a" primary should render the correct output when disabled 1`] = `
<DocumentFragment>
  <a
    class="cursor-not-allowed bg-opacity-75 hover:bg-opacity-75 group items-center relative shadow-2xl inline-flex font-display justify-center font-medium transition-colors select-none focus:outline-none focus:ring-2 focus:ring-offset-2 py-2 px-4 text-base bg-blue-600 text-white"
    disabled=""
    href="https://motech-development.github.io/platform/"
  >
    Test
  </a>
</DocumentFragment>
`;

exports[`Button as element "a" primary should render the correct output when size is lg 1`] = `
<DocumentFragment>
  <a
    class="group items-center relative shadow-2xl inline-flex font-display justify-center font-medium transition-colors select-none focus:outline-none focus:ring-2 focus:ring-offset-2 py-4 px-6 text-lg bg-blue-600 text-white hover:bg-blue-700 focus:ring-blue-500"
    href="https://motech-development.github.io/platform/"
  >
    Test
  </a>
</DocumentFragment>
`;

exports[`Button as element "a" primary should render the correct output when size is md 1`] = `
<DocumentFragment>
  <a
    class="group items-center relative shadow-2xl inline-flex font-display justify-center font-medium transition-colors select-none focus:outline-none focus:ring-2 focus:ring-offset-2 py-2 px-4 text-base bg-blue-600 text-white hover:bg-blue-700 focus:ring-blue-500"
    href="https://motech-development.github.io/platform/"
  >
    Test
  </a>
</DocumentFragment>
`;

exports[`Button as element "a" primary should render the correct output when size is none 1`] = `
<DocumentFragment>
  <a
    class="group items-center relative shadow-2xl inline-flex font-display justify-center font-medium transition-colors select-none focus:outline-none focus:ring-2 focus:ring-offset-2 p-0 text-sm bg-blue-600 text-white hover:bg-blue-700 focus:ring-blue-500"
    href="https://motech-development.github.io/platform/"
  >
    Test
  </a>
</DocumentFragment>
`;

exports[`Button as element "a" primary should render the correct output when size is sm 1`] = `
<DocumentFragment>
  <a
    class="group items-center relative shadow-2xl inline-flex font-display justify-center font-medium transition-colors select-none focus:outline-none focus:ring-2 focus:ring-offset-2 py-0 px-2 text-sm bg-blue-600 text-white hover:bg-blue-700 focus:ring-blue-500"
    href="https://motech-development.github.io/platform/"
  >
    Test
  </a>
</DocumentFragment>
`;

exports[`Button as element "a" secondary should render the correct output when block element 1`] = `
<DocumentFragment>
  <a
    class="w-full group items-center relative shadow-2xl inline-flex font-display justify-center font-medium transition-colors select-none focus:outline-none focus:ring-2 focus:ring-offset-2 py-2 px-4 text-base bg-gray-100 text-gray-600 hover:bg-gray-300 focus:ring-gray-500"
    href="https://motech-development.github.io/platform/"
  >
    Test
  </a>
</DocumentFragment>
`;

exports[`Button as element "a" secondary should render the correct output when disabled 1`] = `
<DocumentFragment>
  <a
    class="cursor-not-allowed bg-opacity-75 hover:bg-opacity-75 group items-center relative shadow-2xl inline-flex font-display justify-center font-medium transition-colors select-none focus:outline-none focus:ring-2 focus:ring-offset-2 py-2 px-4 text-base bg-gray-100 text-gray-600"
    disabled=""
    href="https://motech-development.github.io/platform/"
  >
    Test
  </a>
</DocumentFragment>
`;

exports[`Button as element "a" secondary should render the correct output when size is lg 1`] = `
<DocumentFragment>
  <a
    class="group items-center relative shadow-2xl inline-flex font-display justify-center font-medium transition-colors select-none focus:outline-none focus:ring-2 focus:ring-offset-2 py-4 px-6 text-lg bg-gray-100 text-gray-600 hover:bg-gray-300 focus:ring-gray-500"
    href="https://motech-development.github.io/platform/"
  >
    Test
  </a>
</DocumentFragment>
`;

exports[`Button as element "a" secondary should render the correct output when size is md 1`] = `
<DocumentFragment>
  <a
    class="group items-center relative shadow-2xl inline-flex font-display justify-center font-medium transition-colors select-none focus:outline-none focus:ring-2 focus:ring-offset-2 py-2 px-4 text-base bg-gray-100 text-gray-600 hover:bg-gray-300 focus:ring-gray-500"
    href="https://motech-development.github.io/platform/"
  >
    Test
  </a>
</DocumentFragment>
`;

exports[`Button as element "a" secondary should render the correct output when size is none 1`] = `
<DocumentFragment>
  <a
    class="group items-center relative shadow-2xl inline-flex font-display justify-center font-medium transition-colors select-none focus:outline-none focus:ring-2 focus:ring-offset-2 p-0 text-sm bg-gray-100 text-gray-600 hover:bg-gray-300 focus:ring-gray-500"
    href="https://motech-development.github.io/platform/"
  >
    Test
  </a>
</DocumentFragment>
`;

exports[`Button as element "a" secondary should render the correct output when size is sm 1`] = `
<DocumentFragment>
  <a
    class="group items-center relative shadow-2xl inline-flex font-display justify-center font-medium transition-colors select-none focus:outline-none focus:ring-2 focus:ring-offset-2 py-0 px-2 text-sm bg-gray-100 text-gray-600 hover:bg-gray-300 focus:ring-gray-500"
    href="https://motech-development.github.io/platform/"
  >
    Test
  </a>
</DocumentFragment>
`;

exports[`Button as element "a" success should render the correct output when block element 1`] = `
<DocumentFragment>
  <a
    class="w-full group items-center relative shadow-2xl inline-flex font-display justify-center font-medium transition-colors select-none focus:outline-none focus:ring-2 focus:ring-offset-2 py-2 px-4 text-base bg-green-600 text-white hover:bg-green-700 focus:ring-green-500"
    href="https://motech-development.github.io/platform/"
  >
    Test
  </a>
</DocumentFragment>
`;

exports[`Button as element "a" success should render the correct output when disabled 1`] = `
<DocumentFragment>
  <a
    class="cursor-not-allowed bg-opacity-75 hover:bg-opacity-75 group items-center relative shadow-2xl inline-flex font-display justify-center font-medium transition-colors select-none focus:outline-none focus:ring-2 focus:ring-offset-2 py-2 px-4 text-base bg-green-600 text-white"
    disabled=""
    href="https://motech-development.github.io/platform/"
  >
    Test
  </a>
</DocumentFragment>
`;

exports[`Button as element "a" success should render the correct output when size is lg 1`] = `
<DocumentFragment>
  <a
    class="group items-center relative shadow-2xl inline-flex font-display justify-center font-medium transition-colors select-none focus:outline-none focus:ring-2 focus:ring-offset-2 py-4 px-6 text-lg bg-green-600 text-white hover:bg-green-700 focus:ring-green-500"
    href="https://motech-development.github.io/platform/"
  >
    Test
  </a>
</DocumentFragment>
`;

exports[`Button as element "a" success should render the correct output when size is md 1`] = `
<DocumentFragment>
  <a
    class="group items-center relative shadow-2xl inline-flex font-display justify-center font-medium transition-colors select-none focus:outline-none focus:ring-2 focus:ring-offset-2 py-2 px-4 text-base bg-green-600 text-white hover:bg-green-700 focus:ring-green-500"
    href="https://motech-development.github.io/platform/"
  >
    Test
  </a>
</DocumentFragment>
`;

exports[`Button as element "a" success should render the correct output when size is none 1`] = `
<DocumentFragment>
  <a
    class="group items-center relative shadow-2xl inline-flex font-display justify-center font-medium transition-colors select-none focus:outline-none focus:ring-2 focus:ring-offset-2 p-0 text-sm bg-green-600 text-white hover:bg-green-700 focus:ring-green-500"
    href="https://motech-development.github.io/platform/"
  >
    Test
  </a>
</DocumentFragment>
`;

exports[`Button as element "a" success should render the correct output when size is sm 1`] = `
<DocumentFragment>
  <a
    class="group items-center relative shadow-2xl inline-flex font-display justify-center font-medium transition-colors select-none focus:outline-none focus:ring-2 focus:ring-offset-2 py-0 px-2 text-sm bg-green-600 text-white hover:bg-green-700 focus:ring-green-500"
    href="https://motech-development.github.io/platform/"
  >
    Test
  </a>
</DocumentFragment>
`;

exports[`Button as element "a" warning should render the correct output when block element 1`] = `
<DocumentFragment>
  <a
    class="w-full group items-center relative shadow-2xl inline-flex font-display justify-center font-medium transition-colors select-none focus:outline-none focus:ring-2 focus:ring-offset-2 py-2 px-4 text-base bg-yellow-600 text-white hover:bg-yellow-700 focus:ring-yellow-500"
    href="https://motech-development.github.io/platform/"
  >
    Test
  </a>
</DocumentFragment>
`;

exports[`Button as element "a" warning should render the correct output when disabled 1`] = `
<DocumentFragment>
  <a
    class="cursor-not-allowed bg-opacity-75 hover:bg-opacity-75 group items-center relative shadow-2xl inline-flex font-display justify-center font-medium transition-colors select-none focus:outline-none focus:ring-2 focus:ring-offset-2 py-2 px-4 text-base bg-yellow-600 text-white"
    disabled=""
    href="https://motech-development.github.io/platform/"
  >
    Test
  </a>
</DocumentFragment>
`;

exports[`Button as element "a" warning should render the correct output when size is lg 1`] = `
<DocumentFragment>
  <a
    class="group items-center relative shadow-2xl inline-flex font-display justify-center font-medium transition-colors select-none focus:outline-none focus:ring-2 focus:ring-offset-2 py-4 px-6 text-lg bg-yellow-600 text-white hover:bg-yellow-700 focus:ring-yellow-500"
    href="https://motech-development.github.io/platform/"
  >
    Test
  </a>
</DocumentFragment>
`;

exports[`Button as element "a" warning should render the correct output when size is md 1`] = `
<DocumentFragment>
  <a
    class="group items-center relative shadow-2xl inline-flex font-display justify-center font-medium transition-colors select-none focus:outline-none focus:ring-2 focus:ring-offset-2 py-2 px-4 text-base bg-yellow-600 text-white hover:bg-yellow-700 focus:ring-yellow-500"
    href="https://motech-development.github.io/platform/"
  >
    Test
  </a>
</DocumentFragment>
`;

exports[`Button as element "a" warning should render the correct output when size is none 1`] = `
<DocumentFragment>
  <a
    class="group items-center relative shadow-2xl inline-flex font-display justify-center font-medium transition-colors select-none focus:outline-none focus:ring-2 focus:ring-offset-2 p-0 text-sm bg-yellow-600 text-white hover:bg-yellow-700 focus:ring-yellow-500"
    href="https://motech-development.github.io/platform/"
  >
    Test
  </a>
</DocumentFragment>
`;

exports[`Button as element "a" warning should render the correct output when size is sm 1`] = `
<DocumentFragment>
  <a
    class="group items-center relative shadow-2xl inline-flex font-display justify-center font-medium transition-colors select-none focus:outline-none focus:ring-2 focus:ring-offset-2 py-0 px-2 text-sm bg-yellow-600 text-white hover:bg-yellow-700 focus:ring-yellow-500"
    href="https://motech-development.github.io/platform/"
  >
    Test
  </a>
</DocumentFragment>
`;

exports[`Button as element "button" danger should render the correct output when block element 1`] = `
<DocumentFragment>
  <button
    class="w-full group items-center relative shadow-2xl inline-flex font-display justify-center font-medium transition-colors select-none focus:outline-none focus:ring-2 focus:ring-offset-2 py-2 px-4 text-base bg-red-600 text-white hover:bg-red-700 focus:ring-red-500"
    type="button"
  >
    Test
  </button>
</DocumentFragment>
`;

exports[`Button as element "button" danger should render the correct output when disabled 1`] = `
<DocumentFragment>
  <button
    class="cursor-not-allowed bg-opacity-75 hover:bg-opacity-75 group items-center relative shadow-2xl inline-flex font-display justify-center font-medium transition-colors select-none focus:outline-none focus:ring-2 focus:ring-offset-2 py-2 px-4 text-base bg-red-600 text-white"
    disabled=""
    type="button"
  >
    Test
  </button>
</DocumentFragment>
`;

exports[`Button as element "button" danger should render the correct output when size is lg 1`] = `
<DocumentFragment>
  <button
    class="group items-center relative shadow-2xl inline-flex font-display justify-center font-medium transition-colors select-none focus:outline-none focus:ring-2 focus:ring-offset-2 py-4 px-6 text-lg bg-red-600 text-white hover:bg-red-700 focus:ring-red-500"
    type="button"
  >
    Test
  </button>
</DocumentFragment>
`;

exports[`Button as element "button" danger should render the correct output when size is md 1`] = `
<DocumentFragment>
  <button
    class="group items-center relative shadow-2xl inline-flex font-display justify-center font-medium transition-colors select-none focus:outline-none focus:ring-2 focus:ring-offset-2 py-2 px-4 text-base bg-red-600 text-white hover:bg-red-700 focus:ring-red-500"
    type="button"
  >
    Test
  </button>
</DocumentFragment>
`;

exports[`Button as element "button" danger should render the correct output when size is none 1`] = `
<DocumentFragment>
  <button
    class="group items-center relative shadow-2xl inline-flex font-display justify-center font-medium transition-colors select-none focus:outline-none focus:ring-2 focus:ring-offset-2 p-0 text-sm bg-red-600 text-white hover:bg-red-700 focus:ring-red-500"
    type="button"
  >
    Test
  </button>
</DocumentFragment>
`;

exports[`Button as element "button" danger should render the correct output when size is sm 1`] = `
<DocumentFragment>
  <button
    class="group items-center relative shadow-2xl inline-flex font-display justify-center font-medium transition-colors select-none focus:outline-none focus:ring-2 focus:ring-offset-2 py-0 px-2 text-sm bg-red-600 text-white hover:bg-red-700 focus:ring-red-500"
    type="button"
  >
    Test
  </button>
</DocumentFragment>
`;

exports[`Button as element "button" primary should render the correct output when block element 1`] = `
<DocumentFragment>
  <button
    class="w-full group items-center relative shadow-2xl inline-flex font-display justify-center font-medium transition-colors select-none focus:outline-none focus:ring-2 focus:ring-offset-2 py-2 px-4 text-base bg-blue-600 text-white hover:bg-blue-700 focus:ring-blue-500"
    type="button"
  >
    Test
  </button>
</DocumentFragment>
`;

exports[`Button as element "button" primary should render the correct output when disabled 1`] = `
<DocumentFragment>
  <button
    class="cursor-not-allowed bg-opacity-75 hover:bg-opacity-75 group items-center relative shadow-2xl inline-flex font-display justify-center font-medium transition-colors select-none focus:outline-none focus:ring-2 focus:ring-offset-2 py-2 px-4 text-base bg-blue-600 text-white"
    disabled=""
    type="button"
  >
    Test
  </button>
</DocumentFragment>
`;

exports[`Button as element "button" primary should render the correct output when size is lg 1`] = `
<DocumentFragment>
  <button
    class="group items-center relative shadow-2xl inline-flex font-display justify-center font-medium transition-colors select-none focus:outline-none focus:ring-2 focus:ring-offset-2 py-4 px-6 text-lg bg-blue-600 text-white hover:bg-blue-700 focus:ring-blue-500"
    type="button"
  >
    Test
  </button>
</DocumentFragment>
`;

exports[`Button as element "button" primary should render the correct output when size is md 1`] = `
<DocumentFragment>
  <button
    class="group items-center relative shadow-2xl inline-flex font-display justify-center font-medium transition-colors select-none focus:outline-none focus:ring-2 focus:ring-offset-2 py-2 px-4 text-base bg-blue-600 text-white hover:bg-blue-700 focus:ring-blue-500"
    type="button"
  >
    Test
  </button>
</DocumentFragment>
`;

exports[`Button as element "button" primary should render the correct output when size is none 1`] = `
<DocumentFragment>
  <button
    class="group items-center relative shadow-2xl inline-flex font-display justify-center font-medium transition-colors select-none focus:outline-none focus:ring-2 focus:ring-offset-2 p-0 text-sm bg-blue-600 text-white hover:bg-blue-700 focus:ring-blue-500"
    type="button"
  >
    Test
  </button>
</DocumentFragment>
`;

exports[`Button as element "button" primary should render the correct output when size is sm 1`] = `
<DocumentFragment>
  <button
    class="group items-center relative shadow-2xl inline-flex font-display justify-center font-medium transition-colors select-none focus:outline-none focus:ring-2 focus:ring-offset-2 py-0 px-2 text-sm bg-blue-600 text-white hover:bg-blue-700 focus:ring-blue-500"
    type="button"
  >
    Test
  </button>
</DocumentFragment>
`;

exports[`Button as element "button" secondary should render the correct output when block element 1`] = `
<DocumentFragment>
  <button
    class="w-full group items-center relative shadow-2xl inline-flex font-display justify-center font-medium transition-colors select-none focus:outline-none focus:ring-2 focus:ring-offset-2 py-2 px-4 text-base bg-gray-100 text-gray-600 hover:bg-gray-300 focus:ring-gray-500"
    type="button"
  >
    Test
  </button>
</DocumentFragment>
`;

exports[`Button as element "button" secondary should render the correct output when disabled 1`] = `
<DocumentFragment>
  <button
    class="cursor-not-allowed bg-opacity-75 hover:bg-opacity-75 group items-center relative shadow-2xl inline-flex font-display justify-center font-medium transition-colors select-none focus:outline-none focus:ring-2 focus:ring-offset-2 py-2 px-4 text-base bg-gray-100 text-gray-600"
    disabled=""
    type="button"
  >
    Test
  </button>
</DocumentFragment>
`;

exports[`Button as element "button" secondary should render the correct output when size is lg 1`] = `
<DocumentFragment>
  <button
    class="group items-center relative shadow-2xl inline-flex font-display justify-center font-medium transition-colors select-none focus:outline-none focus:ring-2 focus:ring-offset-2 py-4 px-6 text-lg bg-gray-100 text-gray-600 hover:bg-gray-300 focus:ring-gray-500"
    type="button"
  >
    Test
  </button>
</DocumentFragment>
`;

exports[`Button as element "button" secondary should render the correct output when size is md 1`] = `
<DocumentFragment>
  <button
    class="group items-center relative shadow-2xl inline-flex font-display justify-center font-medium transition-colors select-none focus:outline-none focus:ring-2 focus:ring-offset-2 py-2 px-4 text-base bg-gray-100 text-gray-600 hover:bg-gray-300 focus:ring-gray-500"
    type="button"
  >
    Test
  </button>
</DocumentFragment>
`;

exports[`Button as element "button" secondary should render the correct output when size is none 1`] = `
<DocumentFragment>
  <button
    class="group items-center relative shadow-2xl inline-flex font-display justify-center font-medium transition-colors select-none focus:outline-none focus:ring-2 focus:ring-offset-2 p-0 text-sm bg-gray-100 text-gray-600 hover:bg-gray-300 focus:ring-gray-500"
    type="button"
  >
    Test
  </button>
</DocumentFragment>
`;

exports[`Button as element "button" secondary should render the correct output when size is sm 1`] = `
<DocumentFragment>
  <button
    class="group items-center relative shadow-2xl inline-flex font-display justify-center font-medium transition-colors select-none focus:outline-none focus:ring-2 focus:ring-offset-2 py-0 px-2 text-sm bg-gray-100 text-gray-600 hover:bg-gray-300 focus:ring-gray-500"
    type="button"
  >
    Test
  </button>
</DocumentFragment>
`;

exports[`Button as element "button" success should render the correct output when block element 1`] = `
<DocumentFragment>
  <button
    class="w-full group items-center relative shadow-2xl inline-flex font-display justify-center font-medium transition-colors select-none focus:outline-none focus:ring-2 focus:ring-offset-2 py-2 px-4 text-base bg-green-600 text-white hover:bg-green-700 focus:ring-green-500"
    type="button"
  >
    Test
  </button>
</DocumentFragment>
`;

exports[`Button as element "button" success should render the correct output when disabled 1`] = `
<DocumentFragment>
  <button
    class="cursor-not-allowed bg-opacity-75 hover:bg-opacity-75 group items-center relative shadow-2xl inline-flex font-display justify-center font-medium transition-colors select-none focus:outline-none focus:ring-2 focus:ring-offset-2 py-2 px-4 text-base bg-green-600 text-white"
    disabled=""
    type="button"
  >
    Test
  </button>
</DocumentFragment>
`;

exports[`Button as element "button" success should render the correct output when size is lg 1`] = `
<DocumentFragment>
  <button
    class="group items-center relative shadow-2xl inline-flex font-display justify-center font-medium transition-colors select-none focus:outline-none focus:ring-2 focus:ring-offset-2 py-4 px-6 text-lg bg-green-600 text-white hover:bg-green-700 focus:ring-green-500"
    type="button"
  >
    Test
  </button>
</DocumentFragment>
`;

exports[`Button as element "button" success should render the correct output when size is md 1`] = `
<DocumentFragment>
  <button
    class="group items-center relative shadow-2xl inline-flex font-display justify-center font-medium transition-colors select-none focus:outline-none focus:ring-2 focus:ring-offset-2 py-2 px-4 text-base bg-green-600 text-white hover:bg-green-700 focus:ring-green-500"
    type="button"
  >
    Test
  </button>
</DocumentFragment>
`;

exports[`Button as element "button" success should render the correct output when size is none 1`] = `
<DocumentFragment>
  <button
    class="group items-center relative shadow-2xl inline-flex font-display justify-center font-medium transition-colors select-none focus:outline-none focus:ring-2 focus:ring-offset-2 p-0 text-sm bg-green-600 text-white hover:bg-green-700 focus:ring-green-500"
    type="button"
  >
    Test
  </button>
</DocumentFragment>
`;

exports[`Button as element "button" success should render the correct output when size is sm 1`] = `
<DocumentFragment>
  <button
    class="group items-center relative shadow-2xl inline-flex font-display justify-center font-medium transition-colors select-none focus:outline-none focus:ring-2 focus:ring-offset-2 py-0 px-2 text-sm bg-green-600 text-white hover:bg-green-700 focus:ring-green-500"
    type="button"
  >
    Test
  </button>
</DocumentFragment>
`;

exports[`Button as element "button" warning should render the correct output when block element 1`] = `
<DocumentFragment>
  <button
    class="w-full group items-center relative shadow-2xl inline-flex font-display justify-center font-medium transition-colors select-none focus:outline-none focus:ring-2 focus:ring-offset-2 py-2 px-4 text-base bg-yellow-600 text-white hover:bg-yellow-700 focus:ring-yellow-500"
    type="button"
  >
    Test
  </button>
</DocumentFragment>
`;

exports[`Button as element "button" warning should render the correct output when disabled 1`] = `
<DocumentFragment>
  <button
    class="cursor-not-allowed bg-opacity-75 hover:bg-opacity-75 group items-center relative shadow-2xl inline-flex font-display justify-center font-medium transition-colors select-none focus:outline-none focus:ring-2 focus:ring-offset-2 py-2 px-4 text-base bg-yellow-600 text-white"
    disabled=""
    type="button"
  >
    Test
  </button>
</DocumentFragment>
`;

exports[`Button as element "button" warning should render the correct output when size is lg 1`] = `
<DocumentFragment>
  <button
    class="group items-center relative shadow-2xl inline-flex font-display justify-center font-medium transition-colors select-none focus:outline-none focus:ring-2 focus:ring-offset-2 py-4 px-6 text-lg bg-yellow-600 text-white hover:bg-yellow-700 focus:ring-yellow-500"
    type="button"
  >
    Test
  </button>
</DocumentFragment>
`;

exports[`Button as element "button" warning should render the correct output when size is md 1`] = `
<DocumentFragment>
  <button
    class="group items-center relative shadow-2xl inline-flex font-display justify-center font-medium transition-colors select-none focus:outline-none focus:ring-2 focus:ring-offset-2 py-2 px-4 text-base bg-yellow-600 text-white hover:bg-yellow-700 focus:ring-yellow-500"
    type="button"
  >
    Test
  </button>
</DocumentFragment>
`;

exports[`Button as element "button" warning should render the correct output when size is none 1`] = `
<DocumentFragment>
  <button
    class="group items-center relative shadow-2xl inline-flex font-display justify-center font-medium transition-colors select-none focus:outline-none focus:ring-2 focus:ring-offset-2 p-0 text-sm bg-yellow-600 text-white hover:bg-yellow-700 focus:ring-yellow-500"
    type="button"
  >
    Test
  </button>
</DocumentFragment>
`;

exports[`Button as element "button" warning should render the correct output when size is sm 1`] = `
<DocumentFragment>
  <button
    class="group items-center relative shadow-2xl inline-flex font-display justify-center font-medium transition-colors select-none focus:outline-none focus:ring-2 focus:ring-offset-2 py-0 px-2 text-sm bg-yellow-600 text-white hover:bg-yellow-700 focus:ring-yellow-500"
    type="button"
  >
    Test
  </button>
</DocumentFragment>
`;
